### PR TITLE
APIS-7217 - Generate GraphQL schemas.

### PR DESCRIPTION
**Pull request by gannett-ggreer**

### CODE REVIEW

File: generate/avro.go [hunk 1]
```diff
@@ -392,6 +392,7 @@ func convertToAvroType(cfg avroConfig, expr ast.Expr, name string, nullable bool
 		default: // Another go type
 			n, err := parseGoStruct(t.Name, cfg.dir)
 			if err != nil {
+				//panic(fmt.Errorf("failed to parse Go struct: %v", err))
```
Comment (mmandolesi-g) [inline generate/avro.go:395]: Can this be deleted?

Comment (gannett-ggreer) [reply generate/avro.go:395]: We could.  I left it because it was a useful debugging tool one time when I was getting `"unknown"` returned unexpectedly.

**Review by mmandolesi-g — APPROVED** [review]
This one is probably a good one to walk through with the team.

File: generate/graphql.go [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+
+	if !filepath.IsAbs(schemaPath) {
+		schemaPath = filepath.Join(filepath.Dir(args.SchemaPath), schemaPath)
+	}
+	schema, err := jsonschema.SchemaFromFile(schemaPath, name)
+	if err != nil {
+		return err
+	}
+
+	// Interface.
```
Comment (liamphmurphy) [inline generate/graphql.go:94]: Not sure I understand what this comment means.

File: generate/graphql.go [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+				obj.rootStruct.gqlExtractedField.name = obj.rootStruct.gqlExtractedField.jsonName
+				obj.rootStruct.buildType = "interface"
+			}
+		}
+		if err := obj.write(gfile); err != nil {
+			return fmt.Errorf("failed to write allOf struct: %w", err)
+		}
+	}
+
+	// Implementations.
```
Comment (liamphmurphy) [inline generate/graphql.go:169]: Okay, seeing this comment makes the `Interface` comment above make more sense. But I think the comments could still be a bit more descriptive?

Comment (gannett-ggreer) [reply generate/graphql.go]: Added some more detail to the comments.

**Comment by gannett-ggreer** [comment]
For the case of a schema with 0 AllOf, 1 OneOf, and 0 properties, added a minor fix to not consider the generated type an implementation of an interface (since there is no AllOf).
https://github.com/GannettDigital/jstransform/compare/ccb6952cee16186dc9a654640729dd9adbc347b9..5b7ba582f72720010c8e9d3f17b27e879c3bf731?w=1

**Review by liamphmurphy — APPROVED** [review]

**Pull request merged**